dali/operators/generic/slice/slice_attr.h

```diff
@@ -33,13 +35,30 @@
 
 namespace dali {
 
+namespace detail {
+
+enum class AxesMode : uint8_t {
+  Axes, AxisNames, AllAxes
+};
+
+/**
+ * @brief Gets a SmallVector with the final axes for a particular sample, given the current config
+ */
+SmallVector<int, 8> GetAxes(AxesMode axes_mode, const TensorLayout& shape_layout,
```

Comment:
I think this function is a bit odd.
1. The argument selector is a it strange on its own.
2. The user needs to obtain the axis names (and pass them) and make sure to acquire the newest version.

In my opinion, this utility does too little. I'd rather have a class with an interface resembling this:

```C++
class AxisArgs {
  enum Flags : unsigned {
    AllowEmpty = 0b0001,
    AllowMultiple = 0b0010,
    AllIfEmpty = 0b0100,
    AllowNegative = 0b1000,
    Default = AllowEmpty | AllowMultiple | AllIfEmpty | AllowNegative
  };
  AxisArgs(const OpSpec &spec, const char *index_arg, const char *name_arg, unsigned flags = Default);

  SmallVector<int, 8> Get(int ndim, const OpSpec &spec, const ArgumentWorkspace &ws);
};
```